From 28ce9f8378f50d2ad4380f51aaa13aa10f828b5e Mon Sep 17 00:00:00 2001
From: allen yan <yanwei@marvell.com>
Date: Tue, 27 Jun 2017 19:06:48 +0800
Subject: [PATCH 1046/1345] clk: mvebu: armada-37xx-periph: add support for
 pcie gate clock

commit  3edeb0cba46443f6943db9542015a907172d5e6c from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Currently the clock driver doesn't support pcie gate clock,
  it will make pcie doesn't work.
- This patch adds support for pcie gate clcok.

Change-Id: If2a60932e7438a09af1b5fceccf06ad12f0bbcf0
Signed-off-by: allen yan <yanwei@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/40891
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/mvebu/armada-37xx-periph.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/clk/mvebu/armada-37xx-periph.c b/drivers/clk/mvebu/armada-37xx-periph.c
index 15aac40..6751e4a 100644
--- a/drivers/clk/mvebu/armada-37xx-periph.c
+++ b/drivers/clk/mvebu/armada-37xx-periph.c
@@ -252,6 +252,8 @@ struct clk_divider rate_##_name = {		\
 PERIPH_CLK_FULL_DD(sdio, 11, 14, DIV_SEL0, DIV_SEL0, 3, 6);
 PERIPH_CLK_FULL_DD(usb32_usb2_sys, 16, 16, DIV_SEL0, DIV_SEL0, 9, 12);
 PERIPH_CLK_FULL_DD(usb32_ss_sys, 17, 18, DIV_SEL0, DIV_SEL0, 15, 18);
+static PERIPH_GATE(pcie, 14);
+
 
 static struct clk_periph_data data_sb[] = {
 	REF_CLK_MUX_DD(gbe_50),
@@ -267,6 +269,7 @@ struct clk_divider rate_##_name = {		\
 	REF_CLK_FULL_DD(sdio),
 	REF_CLK_FULL_DD(usb32_usb2_sys),
 	REF_CLK_FULL_DD(usb32_ss_sys),
+	REF_CLK_GATE(pcie, "xtal"),
 	{ },
 };
 
-- 
1.7.9.5

